DirectProgramming/DPC++FPGA/ReferenceDesigns/gunzip/README.md

```diff
@@ -0,0 +1,281 @@
+     ```
+   mkdir build
+   cd build
+   ```
+   To compile for the Intel® PAC with Intel Arria® 10 GX FPGA, run `cmake` using the command:  
+    ```
+    cmake -G "NMake Makefiles" ..
+   ```
+   Alternatively, to compile for the Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX), run `cmake` using the command:
+   ```
+   cmake -G "NMake Makefiles" .. -DFPGA_BOARD=intel_s10sx_pac:pac_s10
+   ```
+
+2. Compile the design through the generated `Makefile`. The following build targets are provided, matching the recommended development flow:
+   * Compile for emulation (fast compile time, targets emulated FPGA device): 
+     ```
+     nmake fpga_emu
+     ```
+   * Generate the optimization report: 
+     ```
+     nmake report
+     ``` 
+   * An FPGA hardware target is not provided on Windows*. 
+
+*Note:* The Intel® PAC with Intel Arria® 10 GX FPGA and Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX) do not yet support Windows*. Compiling to FPGA hardware on Windows* requires a third-party or custom Board Support Package (BSP) with Windows* support.
+
+### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this Reference Design in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [Intel® oneAPI DPC++ FPGA Workflows on Third-Party IDEs](https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-fpga-workflow-on-ide)
+
+## Running the Reference Design
+
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU).
+     ```
+     ./gunzip.fpga_emu    (Linux)
+     gunzip.fpga_emu.exe  (Windows)
+     ```
+
+2. Run the sample on the FPGA device.
+     ```
+     ./gunzip.fpga        (Linux)
+     ```
+
+### Example of Output
+You should see output similar to the following in the console:
+```
+>>>>> Uncompressed File Test <<<<<
+Decompressing '../data/uncompressed.gz' 1 time
+Launching kernels for run 0
+All kernels have finished for run 0
+>>>>> Uncompressed File Test: PASSED <<<<<
+
+>>>>> Statically Compressed File Test <<<<<
+Decompressing '../data/static_compressed.gz' 1 time
+Launching kernels for run 0
+All kernels have finished for run 0
+>>>>> Statically Compressed File Test: PASSED <<<<<
+
+>>>>> Dynamically Compressed File Test <<<<<
+Decompressing '../data/dynamic_compressed.gz' 1 time
+Launching kernels for run 0
+All kernels have finished for run 0
+>>>>> Dynamically Compressed File Test: PASSED <<<<<
+
+>>>>> Throughput Test <<<<<
+Decompressing '../data/tp_test.gz' 5 times
+Launching kernels for run 0
+All kernels have finished for run 0
+Launching kernels for run 1
+All kernels have finished for run 1
+Launching kernels for run 2
+All kernels have finished for run 2
+Launching kernels for run 3
+All kernels have finished for run 3
+Launching kernels for run 4
+All kernels have finished for run 4
+Execution time: 7.12956 ms
+Output Throughput: 561.045 MB/s
+Compression Ratio: 1021.26
```

Comment:
Maybe this should be expressed explicitly as a ratio (xxx:1)?

Also, is this number right?  That file is compressed 1021:1 ???

Reply:
I can switch to a percentage, or use a ratio. What do you think?
and yes. It is a max throughput test.